providers/implementations/ciphers/cipher_chacha20_poly1305.c.in

```diff
@@ -113,7 +113,8 @@ static int chacha20_poly1305_get_ctx_params(void *vctx, OSSL_PARAM params[])
     PROV_CHACHA20_POLY1305_CTX *ctx = (PROV_CHACHA20_POLY1305_CTX *)vctx;
     struct chacha20_poly1305_get_ctx_params_st p;
 
-    p = chacha20_poly1305_get_ctx_params_decoder(params);
+    if (ctx == NULL || !chacha20_poly1305_get_ctx_params_decoder(params, &p))
```

Comment:
ctx NULL checking seems unnecesary?

Reply:
It was deemed desirable to consistently check for this in an earlier PR I did.  Thus, I've added it throughout.  What was there before was a real hodge podge.

Reply:
Looks like EVP_CipherInit_ex2() sets ctx->algctx to NON NULL or it fails.
It would be NULL after EVP_CIPHER_CTX_new() or EVP_CIPHER_CTX_reset(). What is the case this check is needed for?